public/javascripts/SVLabel/src/SVLabel/label/LabelContainer.js

```diff
@@ -109,12 +179,24 @@ function LabelContainer($) {
         var match = this.findLabelByTempId(tempId);
 
         // Label with this id doesn't exist in currentCanvasLabels
-        if(otherLabels.length === currentCanvasLabels.length){
-            currentCanvasLabels.push(match);
+        if(otherLabels.length === this.getCurrentLabels().length){
```

Comment:
Can you add some comments to this function to make it a little more clear what's happening? It probably should have had comments before you made edits, but now it looks overwhelming and really needs comments.

Reply:
yup